### PR TITLE
feat: Implement delete expired index job

### PR DESCRIPTION
**Pull request by highpon**
### Description
Delete expired index from Agent using k8s Job. 
The user sets the `index_id` in config and deletes the index.

This feature is currently implementing.

### Related Issue
close: https://github.com/vdaas/vald/issues/2647

### Versions

- Vald Version: v1.7.13
- Go Version: v1.23.2
- Rust Version: v1.81.0
- Docker Version: v27.3.1
- Kubernetes Version: v1.31.1
- Helm Version: v3.16.2
- NGT Version: v2.2.4
- Faiss Version: v1.9.0

### Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

## Release Notes

- **New Features**
  - Introduced a new index deletion job and associated configurations, including a Kubernetes CronJob and ConfigMap for managing deletion operations.
  - Added new GitHub Actions workflows for building and scanning Docker images related to index deletion.

- **Improvements**
  - Enhanced spell-checking configuration by ignoring specific paths.
  - Improved Docker image build process with new targets and configurations.
  - Added comprehensive configuration details for the index deletion service, including logging, server settings, and observability options.

- **Bug Fixes**
  - Minor formatting corrections in workflow files for consistency.

### CODE REVIEW

**Comment by kpango** [comment]
Thank you for your contribution to Vald, and I am very happy that you tried to create the DeleteExpired Index Job.



After reading the code, you understood about Docker and Workflow auto-generation, and you wrote the code in Vald's typical way.



Meanwhile, as mentioned in the original Issue, we think it would be better to use Remove or RemoveByTimestamp to remove items.

So it would be good if you can change the implementation to use Remove or RemoveByTimestamp in Vald.proto without adding the DeleteIndex RPC in Agent.proto.



I apologize if the information may not be enough.

Vald.proto: Vald Gateways (LB, Filter, Mirror) and Agents (Faiss, NGT, Rust(future work)) implement this.

Agent.proto: implemented by Agents (Faiss, NGT, Rust(future work)).

This means the basic CRUD processing has all been standardized in Vald.proto, so you can use Vald.Proto's Remove or RemoveByTimestamp.

**Comment by highpon** [comment]
@kpango 
Thank you very much!
I was mistaken, I will fix it to be implemented using the RPC already provided in Vald.proto.

**Comment by highpon** [comment]
@kpango 
Fixed implementation using Vald.proto's Remove rpc!
Ready for review. Please review when you have time!

**Review by vankichi — APPROVED** [review]
LGTM :) Thanks for your great job :tada:

File: cmd/index/job/deletion/main.go
Comment (kpango) [inline cmd/index/job/deletion/main.go:18]: please use internal/log instead of official logger.

**Review by kpango — CHANGES_REQUESTED** [review]
Overall, it looks good, but I commented that there were a few package references that were different from other components.

File: pkg/index/job/deletion/service/deleter.go
Comment (highpon) [reply pkg/index/job/deletion/service/deleter.go]: Thanks for your review!
fix: https://github.com/vdaas/vald/pull/2702/commits/c7172a12d68e433c382553d7c5dbfbcc030f1e10

File: cmd/index/job/deletion/main.go
Comment (highpon) [reply cmd/index/job/deletion/main.go]: Thanks for your review!
fix: https://github.com/vdaas/vald/commit/c7172a12d68e433c382553d7c5dbfbcc030f1e10

**Comment by highpon** [comment]
@kpango 
I fixed all your points.
When you have time, please check review!

**Review by kpango — APPROVED** [review]
LGTM, Thank you for your Great Contribution.
We needs 2 approvals before merging, our team member will review this PR in few days.

**Comment by kmrmt** [comment]
https://github.com/all-contributors please add @highpon for code

**Comment by kmrmt** [comment]
@all-contributors please add @highpon for code

**Pull request merged**
